server/src/test/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitorTests.java

```diff
@@ -48,30 +48,8 @@
 public class DiskThresholdMonitorTests extends ESAllocationTestCase {
 
     public void testMarkFloodStageIndicesReadOnly() {
-        AllocationService allocation = createAllocationService(Settings.builder()
-            .put("cluster.routing.allocation.node_concurrent_recoveries", 10).build());
         Settings settings = Settings.EMPTY;
-        MetaData metaData = MetaData.builder()
-            .put(IndexMetaData.builder("test").settings(settings(Version.CURRENT)
-                .put("index.routing.allocation.require._id", "node2")).numberOfShards(1).numberOfReplicas(0))
-            .put(IndexMetaData.builder("test_1").settings(settings(Version.CURRENT)
-                .put("index.routing.allocation.require._id", "node1")).numberOfShards(1).numberOfReplicas(0))
-            .put(IndexMetaData.builder("test_2").settings(settings(Version.CURRENT)
-                .put("index.routing.allocation.require._id", "node1")).numberOfShards(1).numberOfReplicas(0))
-            .build();
-        RoutingTable routingTable = RoutingTable.builder()
-            .addAsNew(metaData.index("test"))
-            .addAsNew(metaData.index("test_1"))
-            .addAsNew(metaData.index("test_2"))
-            .build();
-        ClusterState clusterState = ClusterState.builder(ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
-            .metaData(metaData).routingTable(routingTable).build();
-        logger.info("adding two nodes and performing rerouting");
-        clusterState = ClusterState.builder(clusterState).nodes(DiscoveryNodes.builder().add(newNode("node1"))
-            .add(newNode("node2"))).build();
-        clusterState = allocation.reroute(clusterState, "reroute");
-        logger.info("start primary shard");
-        clusterState = allocation.applyStartedShards(clusterState, clusterState.getRoutingNodes().shardsWithState(INITIALIZING));
+        ClusterState clusterState = bootstrapCluster();
         ClusterState finalState = clusterState;
```

Comment:
No need for `finalState` any more, it's always equal to `clusterState` (which is now effectively final).